src/enclave/Enclave/IntegrityUtils.cpp

```diff
@@ -0,0 +1,210 @@
+#include "IntegrityUtils.h"
+
+void init_log(const tuix::EncryptedBlocks *encrypted_blocks) {
```

Comment:
This function gets one set of `curr_entries` and `past_entries` for each `EncryptedBlocks` object, but could there be multiple of these `EncryptedBlocks` for a single ecall? For example, does this currently happen in `external_sort()` since it should be taking in a concatenated `EncryptedBlocks` object, which could have multiple `curr_entries` and `past_entries`?

Reply:
`curr_entries` and `past_entries` are each a list. In the case that a ecall takes in a concatenated `EncryptedBlocks` object, the concatenation functionality in Scala will combine the `curr_entries` objects from the multiple `EncryptedBlocks` that are concatenated. Same thing happens with `past_entries`.

Essentially, the `EncryptedBlocks` output of every ecall will have one element each in the `curr_entries` list, but if the output of two ecalls, two `EncryptedBlocks`, are combined as one `EncryptedBlocks` input to a subsequent ecall, the one element in each `curr_entries` list will also be combined into a two element `curr_entries` list belonging to one `EncryptedBlocks`.